sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Rules/UriShouldNotBeHardcodedTest.cs

```diff
@@ -27,12 +27,24 @@ namespace SonarAnalyzer.UnitTest.Rules
     [TestClass]
     public class UriShouldNotBeHardcodedTest
     {
-        [TestMethod]
         [TestCategory("Rule")]
-        public void UriShouldNotBeHardcoded()
+        [DataTestMethod]
+        [DataRow("3.0.20105.1", "2.0.4", "2.0.3")]
+        [DataRow(AssemblyReference.NuGetInfo.LatestVersion, AssemblyReference.NuGetInfo.LatestVersion, AssemblyReference.NuGetInfo.LatestVersion)]
+        public void UriShouldNotBeHardcoded(string aspNetMvcVersion, string aspNetCoreMvcVersion, string aspNetCoreRoutingVersion)
         {
             Verifier.VerifyAnalyzer(@"TestCases\UriShouldNotBeHardcoded.cs",
-                new UriShouldNotBeHardcoded());
+                new UriShouldNotBeHardcoded(),
+                additionalReferences: new AssemblyReference[] {
+                    AssemblyReference.FromFramework("System.Web.dll"),
+                    AssemblyReference.FromNuGet("System.Web.Mvc.dll", "Microsoft.AspNet.Mvc", aspNetMvcVersion),
+                    // for VirtualFileResult
+                    AssemblyReference.FromNuGet("Microsoft.AspNetCore.Mvc.Core.dll", "Microsoft.AspNetCore.Mvc.Core", aspNetCoreMvcVersion),
```

Comment:
[minor][optional] I would have split the test cases into 3 classes, one for each:

- Classic paths
- AspNet Mvc
- AspNet Core Mvc

Reply:
Fair point, in case it fails because of the dependency, it's easier to spot